### PR TITLE
Break runtime in more digestible files

### LINKED ISSUE #309: Break runtime in more digestible files
## Describe the enhancement
In order to unclutter the runtime code we should split it into smaller files. I like I have done for my Governance OS for instance.

### PR DESCRIPTION
**Pull request by shamb0**
fixes #309 

**Pallet refactor structure**

* "runtime/src/pallets_system.rs"
	- frame_system
	- pallet_timestamp
	- pallet_indices
	- pallet_balances
	- pallet_transaction_payment
* "runtime/src/pallets_consensus.rs"
	- pallet_babe
	- pallet_grandpa
	- pallet_authority_discovery
	- pallet_authorship
	- pallet_im_online
	- pallet_session
	- pallet_poa
	- pallet_membership
	- pallet_offences
* "runtime/src/pallets_governance.rs"
	- pallet_grants
	- pallet_collective
	- pallet_membership
	- pallet_mandate
	- pallet_amendments
	- pallet_reserve
	- pallet_scheduler
* "runtime/src/pallets_util.rs"
	- pallet_identity
	- pallet_recovery
	- pallet_utility
	- pallet_proxy
	- pallet_multisig
	- pallet_contracts
* "runtime/src/pallets_nodle.rs"
	- pallet_tcr
	- pallet_root_of_trust
	- pallet_emergency_shutdown
	- pallet_allocations

### CODE REVIEW

File: runtime/src/pallets_governance.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+use sp_core::u32_trait::{_1, _2};
+use sp_runtime::ModuleId;
+pub use sp_runtime::{Perbill, Perquintill};
+
+impl pallet_grants::Config for Runtime {
+    type Event = Event;
+    type Currency = Balances;
+    type CancelOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, FinancialCollective>;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_governance.rs:36]: Let's move this to `util`

File: runtime/src/pallets_consensus.rs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type SwapOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type ResetOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type PrimeOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type MembershipInitialized = TechnicalCommittee;
+    type MembershipChanged = TechnicalCommittee;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_consensus.rs:173]: This should be moved to `governance`

File: runtime/src/pallets_governance.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, TechnicalCollective>;
+    type SwapOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, TechnicalCollective>;
+    type ResetOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, TechnicalCollective>;
+    type PrimeOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, TechnicalCollective>;
+    type MembershipInitialized = Allocations;
+    type MembershipChanged = Allocations;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_governance.rs:128]: Let's move this to `Nodle`

File: runtime/src/pallets_governance.rs [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+impl pallet_scheduler::Config for Runtime {
+    type Event = Event;
+    type Origin = Origin;
+    type PalletsOrigin = OriginCaller;
+    type Call = Call;
+    type MaximumWeight = MaximumSchedulerWeight;
+    type ScheduleOrigin = frame_system::EnsureRoot<AccountId>;
+    type MaxScheduledPerBlock = MaxScheduledPerBlock;
+    type WeightInfo = pallet_scheduler::weights::SubstrateWeight<Runtime>;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_governance.rs:207]: Probably should go to something like `util`

**Review by ETeissonniere — COMMENTED** [review]
Nice! Some good progress :). I have left a few comments, I think a few things need to be moved here and there

File: runtime/src/pallets_governance.rs (hunk 4 above)
Comment (shamb0) [reply runtime/src/pallets_governance.rs:207]: Done :+1:

File: runtime/src/pallets_governance.rs (hunk 3 above)
Comment (shamb0) [reply runtime/src/pallets_governance.rs:128]: Done :+1:

File: runtime/src/pallets_consensus.rs (hunk 2 above)
Comment (shamb0) [reply runtime/src/pallets_consensus.rs:173]: Done :+1:

File: runtime/src/pallets_governance.rs (hunk 1 above)
Comment (shamb0) [reply runtime/src/pallets_governance.rs:36]: Done :+1:

File: runtime/src/pallets_governance.rs [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+impl pallet_collective::Config<TechnicalCollective> for Runtime {
+    type Origin = Origin;
+    type Proposal = Call;
+    type Event = Event;
+    type MotionDuration = MotionDuration;
+    type MaxProposals = MaxProposals;
+    type WeightInfo = ();
+    type MaxMembers = MaxMembers;
+    type DefaultVote = pallet_collective::PrimeDefaultVote;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_governance.rs:48]: Let's move this down to the `Technical committee` section

File: runtime/src/pallets_governance.rs [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type SwapOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type ResetOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type PrimeOrigin =
+        pallet_collective::EnsureProportionMoreThan<_1, _2, AccountId, RootCollective>;
+    type MembershipInitialized = FinancialCommittee;
+    type MembershipChanged = FinancialCommittee;
+}
```
Comment (ETeissonniere) [inline runtime/src/pallets_governance.rs:93]: Let's move this up to the `Financial committee` section

**Review by ETeissonniere — CHANGES_REQUESTED** [review]

File: runtime/src/pallets_governance.rs (hunk 5 above)
Comment (shamb0) [reply runtime/src/pallets_governance.rs:48]: Thanks Eliott

I hope now I aligned & paired it properly

| committee | pallet_collective | pallet_membership |
| --- | --- | --- |
| Technical | Instance2 | Instance1 |
| Financial | Instance3 | Instance3 |
| Root | Instance4 | Instance4 |

File: runtime/src/pallets_governance.rs (hunk 6 above)
Comment (shamb0) [reply runtime/src/pallets_governance.rs:93]: Done :+1:

**Review by ETeissonniere — APPROVED** [review]
lgtm

**Review by ETeissonniere — APPROVED** [review]

**Pull request merged**